clippy_lints/src/doc.rs

```diff
@@ -258,6 +259,18 @@ fn check_word(cx: &EarlyContext, word: &str, span: Span) {
         s != "_" && !s.contains("\\_") && s.contains('_')
     }
 
+    if let Ok(url) = Url::parse(word) {
```

Comment:
I think it's fine as it is... we have many things in clippy that we should benchmark, we shouldn't optimize before actually doing that. I'd wager the URL parser isn't much worse in perf than us adding a quick check